internal/lintcontext/parse_yaml.go

```diff
@@ -113,6 +114,79 @@ func (l *lintContextImpl) loadObjectsFromHelmChart(dir string) error {
 	return nil
 }
 
+func (l *lintContextImpl) loadObjectsFromTgzHelmChart(tgzFile string) error {
+	metadata := ObjectMetadata{FilePath: tgzFile}
+	renderedFiles, err := l.renderTgzHelmChart(tgzFile)
+	if err != nil {
+		l.invalidObjects = append(l.invalidObjects, InvalidObject{Metadata: metadata, LoadErr: err})
+		return nil
+	}
+	for path, contents := range renderedFiles {
+		// The first element of path will be the same as the last element of tgzFile, because
+		// Helm duplicates it.
+		pathToTemplate := filepath.Join(filepath.Dir(tgzFile), path)
+		if err := l.loadObjectsFromReader(pathToTemplate, strings.NewReader(contents)); err != nil {
+			return errors.Wrapf(err, "loading objects from rendered helm chart %s/%s", tgzFile, pathToTemplate)
+		}
+	}
+	return nil
+}
+
+func (l *lintContextImpl) renderTgzHelmChart(tgzFile string) (map[string]string, error) {
+	// Helm doesn't have great logging behaviour, and can spam stderr, so silence their logging.
+	// TODO: capture these logs.
+	log.SetOutput(nopWriter{})
+	defer log.SetOutput(os.Stderr)
+	chrt, err := loader.LoadFile(tgzFile)
+
+	if err != nil {
+		return nil, err
+	}
+	if err := chrt.Validate(); err != nil {
+		return nil, err
+	}
+
+	valuesIndex := -1
+	for i, f := range chrt.Raw {
+		if f.Name == "values.yaml" {
+			valuesIndex = i
+			break
+		}
+	}
+
+	indexName := filepath.Join(tgzFile, "values.yaml")
+	if valuesIndex == -1 {
+		return nil, errors.Errorf("%s not found", indexName)
+	}
+
+	values, err := l.parseValues(indexName, chrt.Raw[valuesIndex].Data)
+	if err != nil {
+		return nil, errors.Wrap(err, "loading values.yaml file")
+	}
+
+	valuesToRender, err := chartutil.ToRenderValues(chrt, values, chartutil.ReleaseOptions{Name: "test-release", Namespace: "default"}, nil)
+	if err != nil {
+		return nil, err
+	}
+	e := engine.Engine{LintMode: true}
+	rendered, err := e.Render(chrt, valuesToRender)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to render")
+	}
```

Comment:
Can you extract this into a separate function that is reused in `renderHelmChart`?

Reply:
Updated